lib/inspec/resources/sys_info.rb

```diff
@@ -13,13 +13,36 @@ class System < Inspec.resource(1)
       describe sys_info do
         its('hostname') { should eq 'example.com' }
       end
+
+      describe sys_info do
+        its('fqdn') { should eq 'user.example.com' }
+      end
+
     EXAMPLE
 
+    %w{ domain fqdn ip-address short }.each do |opt|
+      define_method(opt.to_sym) do
+        hostname(opt)
+      end
+    end
+
     # returns the hostname of the local system
-    def hostname
+    def hostname(opt = nil)
       os = inspec.os
       if os.linux? || os.darwin?
-        inspec.command("hostname").stdout.chomp
+        opt = case opt
+              when "f", "long", "fqdn", "full"
+                " -f"
+              when "d", "domain"
+                " -d"
+              when "i", "ip-address"
```

Comment:
One thought - it might be better to have this as ip_address. If you're using this to lookup values (as opposed to validate) you need to be able to outright call the method, like this:
```
my_host = sys_info.fqdn
my_ip = sys_info.ip-address # Would need to send a symbol)
```

Reply:
I have a loop above that defines them as methods so they can be accessed that way. I switched it over, I didn't realize ruby doesn't like hyphens.